objstorage/objstorageprovider/sharedobjcat/catalog.go

```diff
@@ -218,28 +218,38 @@ func (c *Catalog) ApplyBatch(b Batch) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	addedObjects := make([]base.DiskFileNum, 0, len(b.ve.NewObjects))
```

Comment:
Do we need this map? Can't we just go over NewObjects to undo the add?

Reply:
Oopsie, fixed!